### PR TITLE
StEvent->StMuDst for sTGC data

### PR DESCRIPTION
**Pull request by jdbrice**
# Adds MuDst support for the sTGC
Additions:
- StMuFttRawHit.h, .cxx
- StMuFttCluster.h, .cxx
- StMuFttPoint.h, .cxx
- StMuFttUtil.h .cxx
- StMuFttCollection.h .cxx

Also changes to the StMuDst (Maker) to add in the sTGC structures.
NOTE: hit <-> cluster <-> point associations are not currently rebuilt, since I dont have a good way to test it yet.
Will add this in a later commit. 

## Test WRITE with
```sh
root4star -b -q -l 'bfc.C( 50, "in, db, MakeEvent, CMuDst,-StrngMuDST", "input.event.root")
```

## Test READ with script: 
https://gist.github.com/jdbrice/fd10e03701802d2853d12af7f917fca9

### CODE REVIEW

File: StRoot/StMuDSTMaker/COMMON/StMuDstMaker.cxx [hunk 1]
```diff
@@ -296,16 +312,16 @@ void StMuDstMaker::zeroArrays()
 void StMuDstMaker::SetStatus(const char *arrType,int status)
 {
 #ifndef __NO_STRANGE_MUDST__
-  static const char *specNames[]={"MuEventAll","StrangeAll","MCAll","EmcAll","PmdAll","FMSAll","FcsAll","TofAll","BTofAll","ETofAll","EpdAll","MTDAll","FgtAll","EztAll",0};  /// dongx, MALisa
+  static const char *specNames[]={"MuEventAll","StrangeAll","MCAll","EmcAll","PmdAll","FMSAll","FcsAll","FttAll","TofAll","BTofAll","ETofAll","EpdAll","MTDAll","FgtAll","EztAll",0};  /// dongx, MALisa
 #else
-  static const char *specNames[]={"MuEventAll",             "MCAll","EmcAll","PmdAll","FMSAll","FcsAll","TofAll","BTofAll","ETofAll","EpdAll","MTDAll","FgtAll","EztAll",0};  /// dongx, MALisa
+  static const char *specNames[]={"MuEventAll",             "MCAll","EmcAll","PmdAll","FMSAll","FcsAll","FcsAll","TofAll","BTofAll","ETofAll","EpdAll","MTDAll","FgtAll","EztAll",0};  /// dongx, MALisa
```
Comment (veprbl) [inline StRoot/StMuDSTMaker/COMMON/StMuDstMaker.cxx]:
```suggestion
  static const char *specNames[]={"MuEventAll",             "MCAll","EmcAll","PmdAll","FMSAll","FcsAll","FttAll","TofAll","BTofAll","ETofAll","EpdAll","MTDAll","FgtAll","EztAll",0};  /// dongx, MALisa
```

Comment (jdbrice) [reply StRoot/StMuDSTMaker/COMMON/StMuDstMaker.cxx]: good catch

Comment (jdbrice) [reply StRoot/StMuDSTMaker/COMMON/StMuDstMaker.cxx]: fixed with ae5f94b

File: StRoot/StMuDSTMaker/COMMON/StMuFttRawHit.h [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    UChar_t mStrip;
+    UChar_t mOrientation;
+
+    // StFttCluster *mCluster;
+    // StFttPoint   *mPoint;
+
+    ClassDef( StMuFttRawHit, 1 );
+};
+
+ostream& operator << ( ostream&, const StMuFttRawHit& hit ); // Printing operator
```
Comment (veprbl) [inline StRoot/StMuDSTMaker/COMMON/StMuFttRawHit.h]: I guess, this is how you fix the ROOT6 build:
```suggestion
std::ostream& operator << ( std::ostream&, const StMuFttRawHit& hit ); // Printing operator
```

Comment (starsdong) [reply StRoot/StMuDSTMaker/COMMON/StMuFttRawHit.h]: Daniel, would you mind taking into account Dima's suggestion and updating the request to fix the ROOT6 build issue? Let us try to get this merged to include it in fast offline?

Comment (jdbrice) [reply StRoot/StMuDSTMaker/COMMON/StMuFttRawHit.h]: thanks 👍

**Review by starsdong — APPROVED** [review]
I looked through the files and they seem to be good.

**Review by akioogawa — APPROVED** [review]

**Pull request merged**